admin/src/react-components/service-editor.js

```diff
@@ -359,11 +404,73 @@ class ConfigurationEditor extends Component {
     }
   }
 
-  renderTree(schema, category, config) {
-    const configurables = getDescriptors(schema[category])
+  renderCategory(schema, category, config) {
+    switch (category) {
+      case "theme":
+        return this.renderThemeSection(schema.theme, config);
+      default:
+        return this.renderTree(schema, category, config);
+    }
+  }
+
+  getFilteredDescriptors(category) {
+    return getDescriptors(category)
       .filter(([, descriptor]) => qs.get("show_internal_configs") !== null || descriptor.internal !== "true")
-      .filter(([, descriptor]) => qs.get("show_deprecated_configs") !== null || descriptor.deprecated !== "true")
-      .map(([path, descriptor]) => this.renderConfigurable(path, descriptor, getConfigValue(config, path)));
+      .filter(([, descriptor]) => qs.get("show_deprecated_configs") !== null || descriptor.deprecated !== "true");
+  }
+
+  /**
+   * Theme Tab Section
+   */
+  renderThemeSection(theme, config) {
+    const configurables = this.getFilteredDescriptors(theme);
+    const getInput = ([path, descriptor]) => this.renderConfigurable(path, descriptor, getConfigValue(config, path));
+
+    return (
+      <form onSubmit={this.onSubmit.bind(this)}>
+        <h3 className="heading-sm mb-24">Nametags</h3>
+        {getInput(configurables[0])}
+        {getInput(configurables[1])}
```

Comment:
Since you know which configurables you need to render (`theme.nametag-color` and `theme.nametag-text-color`) -- it might be better to specify them by `path` rather than by index.

Something like

```ts
const configurable = (path) => {
  const descriptor = getDescriptors( path.split(".")[0] ).find( ([path, desc]) => desc.path === path )[1];
  return this.renderConfigurable(path, descriptor, getConfigValue(config, path));
}

// and in place of getInput, we would write:
configurable("theme.nametag-color")
configurable("theme.nametag-text-color")
...
configurable("theme.themes")
```